### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by jakesokol1**

### CODE REVIEW

File: pkg/controller/apiserver/handlers/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+func TestIncompleteOrInvalidArguments(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	// sample selector arguments (right now, only supports podname and namespace)
+	podName, namespace := "podName", "namespace"
+	// outline test cases with expected behavior
+	testCases := map[string]TestCase{
+		"Responds with empty list given no name and no namespace": {
+			handlerArgs:           []string{"", ""},
+			expectedStatus:  http.StatusOK,
```
Comment (antoninbas) [inline pkg/controller/apiserver/handlers/handler_test.go:83]: is `StatusOK` the right thing to do here? Would you say these are error cases or no?

File: pkg/controller/apiserver/handlers/endpoint/handler.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	Ingress []types.NetworkPolicy `json:"ingress"`
+}
+
+// HandleFunc creates a http.HandlerFunc which uses an AgentNetworkPolicyInfoQuerier
+// to query network policy rules in current agent.
+func HandleFunc(cnpq querier.ControllerNetworkPolicyInfoQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		podName := r.URL.Query().Get("pod")
+		namespace := r.URL.Query().Get("namespace")
+		println("query")
```
Comment (antoninbas) [inline pkg/controller/apiserver/handlers/endpoint/handler.go:38]: let's get into the habit of removing these statements before checking code in
the `println` built-in function is really meant for quick debugging

don't be shy about adding some klog statements, and you can potentially increase the verbosity before checking the code in

File: pkg/controller/apiserver/handlers/endpoint/handler.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"encoding/json"
```
Comment (antoninbas) [inline pkg/controller/apiserver/handlers/endpoint/handler.go:18]: take a look at other files to see how we group imports, that will make the review process more efficient when we submit a PR against upstream antrea eventually

**Review by antoninbas — COMMENTED** [review]
A few early comments. Hopefully you will get an email about this @jakesokol1

File: pkg/controller/apiserver/handlers/handler_test.go (hunk 1 above)
Comment (jakesokol1) [reply pkg/controller/apiserver/handlers/handler_test.go:83]: Maybe http.StatusNoContent would be correct here?

Comment (antoninbas) [reply pkg/controller/apiserver/handlers/handler_test.go:83]: I believe we have been using `http.StatusBadRequest` consistently for such cases

Comment (jakesokol1) [reply pkg/controller/apiserver/handlers/handler_test.go:83]: I have made the changes

File: pkg/controller/apiserver/handlers/endpoint/handler.go (hunk 3 above)
Comment (jakesokol1) [reply pkg/controller/apiserver/handlers/endpoint/handler.go:18]: Will make sure everything is consistent tomorrow morning.

**Pull request merged**